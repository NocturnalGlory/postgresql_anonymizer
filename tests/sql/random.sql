CREATE EXTENSION IF NOT EXISTS anon CASCADE;

SELECT anon.load();

--
-- Generic Types
--

-- zip
SELECT pg_typeof(anon.random_zip()) = 'TEXT'::REGTYPE;


-- string

SELECT pg_typeof(anon.random_string(1)) = 'TEXT'::REGTYPE;
--SELECT anon_string(123456789);


-- Date
SELECT pg_typeof(anon.random_date_between('1900-01-01'::TIMESTAMP WITH TIME ZONE,now())) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
SELECT pg_typeof(anon.random_date_between('0001-01-01'::DATE,'4001-01-01'::DATE)) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
SELECT pg_typeof(anon.random_date()) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;

-- Integer
SELECT pg_typeof(anon.random_int_between(1,3)) = 'INTEGER'::REGTYPE;


-- Phone
SELECT pg_typeof(anon.random_phone('0033')) = 'TEXT'::REGTYPE;
SELECT anon.random_phone(NULL) IS NULL;
SELECT pg_typeof(anon.random_phone()) = 'TEXT'::REGTYPE;

DROP EXTENSION anon CASCADE;
