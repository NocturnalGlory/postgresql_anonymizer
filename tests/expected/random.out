CREATE EXTENSION IF NOT EXISTS anon CASCADE;
SELECT anon.load();
 load 
------
 t
(1 row)

--
-- Generic Types
--
-- zip
SELECT pg_typeof(anon.random_zip()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- string
SELECT pg_typeof(anon.random_string(1)) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--SELECT anon_string(123456789);
-- Date
SELECT pg_typeof(anon.random_date_between('1900-01-01'::TIMESTAMP WITH TIME ZONE,now())) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_date_between('0001-01-01'::DATE,'4001-01-01'::DATE)) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_date()) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Integer
SELECT pg_typeof(anon.random_int_between(1,3)) = 'INTEGER'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Phone
SELECT pg_typeof(anon.random_phone('0033')) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_phone(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_phone()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

DROP EXTENSION anon CASCADE;
